pkg/logentry/stages/util.go

```diff
@@ -78,6 +78,14 @@ func convertDateLayout(predef string, location *time.Location) parser {
 			}
 			return time.Unix(0, i*int64(time.Millisecond)), nil
 		}
+	case "UnixUs":
+		return func(t string) (time.Time, error) {
+			i, err := strconv.ParseInt(t, 10, 64)
+			if err != nil {
+				return time.Time{}, err
+			}
+			return time.Unix(0, i*int64(time.Microsecond)), nil
```

Comment:
I would just use `*1000` instead. This has implicit requirement on durations being in nanosecond, while ms -> us is simple *1000 conversion.

Reply:
should the `time.Millisecond` for the `UnixMs` case be changed to `*1000000` also then? I'm also a bit confused about the implicit requirement here, isn't the `time` package already based on the notion of durations being in nanosecond?

Reply:
> should the `time.Millisecond` for the `UnixMs` case be changed to `*1000000` also then?

That would make sense.

> I'm also a bit confused about the implicit requirement here, isn't the time package already based on the notion of durations being in nanosecond?

Yes it is, and that is the implicit part I'm talking about. Correct expression would be `(time.Duration(i) * time.Microsecond).Nanoseconds()`, which will work even if `time.Duration` starts to use seconds one day.

This is just a tiny nitpick. If you leave it as is, I'm fine with it.